### ISSUE TITLE
fix(vscode): toImportSpecifier() prepends current Windows drive to POSIX absolute paths (v5 mirror of #3836)

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3842]
- Markuplint version: dev branch HEAD (v5 RC)
- Parser lang: N/A (vscode extension server)
- Node.js version: 24
- OS: Windows

## Describe the bug

`vscode/src/server/get-module.ts` on the dev (v5 RC) branch carries the identical Windows-POSIX path bug already fixed for v4 in #3841 (Issue #3836).

`toImportSpecifier()` delegates POSIX absolute paths to Node's `pathToFileURL()`, which on a Windows runtime resolves `/tmp/foo` against the current drive and returns `file:///D:/tmp/foo` instead of `file:///tmp/foo`. The unit tests at `vscode/src/server/get-module.spec.ts:39` and `:43` already encode the correct contract and fail on Windows CI.

## Steps To Reproduce

On Windows:

```sh
git checkout dev
npx vitest run vscode/src/server/get-module.spec.ts
```

Both POSIX-path tests fail with received `'file:///D:/tmp/...'`.

## The current behavior

```
AssertionError: expected 'file:///D:/tmp/markuplint/lib/index.mjs'
                to be 'file:///tmp/markuplint/lib/index.mjs'
```

## The expected behavior

POSIX absolute paths must round-trip: `/tmp/foo.mjs` → `file:///tmp/foo.mjs`, regardless of the host OS.

## Why this is a separate Issue from #3836

- #3836 was scoped to v4 (now fixed via #3841 merged 2026-05-09).
- The dev (v5 RC) branch and v4 are diverged and cannot be merged (per project policy), so v4 fixes do not propagate automatically.
- `vscode/**` is in the `paths:` filter on dev's `.github/workflows/test.yml`, so the bug surfaces on every dev-base PR matrix run; this currently blocks #3840 (file-resolver same root cause on dev).

## Suggested fix

Forward-port #3841's commit `6ac8ce4d3` (and the cspell follow-up `386cb8ab4`) to dev. The vscode source on v4 and dev is byte-identical, so the patch should apply cleanly.

## Tests

The 18 tests in `vscode/src/server/get-module.spec.ts` (after #3841) provide the regression coverage — they need to be the same on dev.
